src/pages/studyView/studyPageHeader/studyViewWarning/StudyViewWarning.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { Alert, Button, Modal, Checkbox, ButtonGroup } from 'react-bootstrap';
+
+export interface IStudyViewWarningBannerProps {
+    message: string;
+}
+
+export interface IStudyViewWarningBannerStates {
+    show: boolean;
+    modalShow: boolean;
+    agreementOne: boolean;
+    agreementTwo: boolean;
+}
+
+export default class StudyViewWarning extends React.Component<
+    IStudyViewWarningBannerProps,
+    IStudyViewWarningBannerStates
+> {
+    constructor(props: any) {
+        super(props);
+
+        this.handleHide = this.handleHide.bind(this);
+        this.handleShow = this.handleShow.bind(this);
+        this.handleDismiss = this.handleDismiss.bind(this);
+        this.handleModalShow = this.handleModalShow.bind(this);
+        this.handleModalHide = this.handleModalHide.bind(this);
+        this.handleChecked = this.handleChecked.bind(this);
+
+        this.state = {

```

Comment:
we don't use React state.  We use mobx observables.   Not a big deal but you should familiarize yourself with how it works.   Look at any other component and you will see see @observable decorator on instance properties.  And you'll see @observer decorator on the class.  This wires up render to observe changes to observable properties exactly as setState does.